File: padre/integration/test_files/test_print_variables.rs
use std::io;

struct TestStructInner<'a> {
    a: String,
    b: &'a str,
}

struct TestStruct<'a> {
    a: bool,
    b: i32,
    c: Vec<String>,
    d: TestStructInner<'a>,
}

fn main() -> io::Result<()> {
    let a: i32 = 42;
    let b = &a;
    let a: f32 = 42.1;
    let a: bool = true;
    let a = "TEST";
    let b = &a;
    let a = "TEST".to_string();
    let b = &a;
    let a: Vec<String> = vec![
        "TEST1".to_string(),
        "TEST2".to_string(),
        "TEST3".to_string(),
    ];
    let a: TestStruct = TestStruct {
        a: true,
        b: 42,
        c: vec!["TEST1".to_string()],
        d: TestStructInner {
            a: "TEST_INNER".to_string(),
            b: "TESTING",
        },
    };

    Ok(())
}

File: padre/integration/test_files/test_print_variables.cpp
#include <string>
#include <vector>

class TestClass {
public:
    std::string a;
    const char* b;
};

struct TestStruct {
    bool a;
    int b;
    std::vector<std::string> c;
    TestClass d;
};

int main() {
    int a = 42;
    int &b = a;
    int *c = &a;
    bool d = true;
    const char* e = "TEST";
    const char* &f = e;
    std::string g = "TEST";
    std::string &h = g;
    std::string *i = &g;
    std::vector<std::string> j = {"TEST1", "TEST2", "TEST3"};
    TestStruct k;
    k.a = true;
    k.b = 42;
    k.c = {"TEST1"};
    TestClass l;
    l.a = "TEST_INNER";
    l.b = "TESTING";
    k.d = l;

    return 0;
}
